src/main/java/org/embulk/output/kintone/KintonePageOutput.java

```diff
@@ -337,25 +308,40 @@ private void upsert(ArrayList<Record> records, ArrayList<UpdateKey> updateKeys)
         updateRecords.clear();
       }
     }
-    if (insertRecords.size() > 0) {
+    if (!insertRecords.isEmpty()) {
       insert(insertRecords);
     }
-    if (updateRecords.size() > 0) {
+    if (!updateRecords.isEmpty()) {
       update(updateRecords);
     }
   }
 
-  private List<Record> getExistingRecordsByUpdateKey(ArrayList<UpdateKey> updateKeys) {
-    String fieldCode = updateKeys.get(0).getField();
+  private List<String> getExistingValuesByUpdateKey(List<UpdateKey> updateKeys) {
+    String fieldCode =
+        updateKeys.stream()
+            .map(UpdateKey::getField)
+            .filter(Objects::nonNull)
+            .findFirst()
+            .orElse(null);
+    if (fieldCode == null) {
+      return Collections.emptyList();
+    }
+    Function<Record, String> fieldValueAsString;
+    FieldType fieldType = getFieldType(fieldCode);
+    if (fieldType == FieldType.SINGLE_LINE_TEXT) {
+      fieldValueAsString = record -> record.getSingleLineTextFieldValue(fieldCode);
+    } else if (fieldType == FieldType.NUMBER) {
+      fieldValueAsString = record -> toString(record.getNumberFieldValue(fieldCode));
```

Comment:
The actual type of the number field value is BigDecimal, but it is converted to String to make it as comparable as possible.